pkg/agent/run.go

```diff
@@ -184,37 +184,61 @@ func gatherData(ctx context.Context, config Config) []*api.DataReading {
 		dataGatherers[dgConfig.Name] = dg
 	}
 
-	// Fetch from all datagatherers
-	now := time.Now()
+	//TODO Change backoff parameters to those desired
+	backOff := backoff.NewExponentialBackOff()
+	backOff.MaxElapsedTime = 30 * time.Second
 	readings := []*api.DataReading{}
-	failedDataGatherers := []string{}
-	for k, dg := range dataGatherers {
-		i, err := dg.Fetch()
-		if err != nil {
-			log.Printf("Error fetching with DataGatherer %q: %s", k, err)
-			failedDataGatherers = append(failedDataGatherers, k)
-			continue
-		}
-
-		log.Printf("Gathered data for %q:\n", k)
+	completedDataGatherers := make(map[string]bool)
```

Comment:
Since you already know the final size of the map (`len(dataGatherers)` in an ideal world), instantiate it with that size `make(map[string]bool, len(dataGatherers))`.